link/xdp.go

```diff
@@ -0,0 +1,64 @@
+package link
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/cilium/ebpf"
+)
+
+// XDPAttachFlags represents how XDP program will be attached to interface.
+type XDPAttachFlags uint32
+
+const (
+	// XDPGenericMode (SKB) links XDP BPF program for drivers which do
+	// not yet support native XDP.
+	XDPGenericMode XDPAttachFlags = 1 << (iota + 1)
+	// XDPDriverMode links XDP BPF program into the driver’s receive path.
+	XDPDriverMode
+	// XDPOffloadMode offloads the entire XDP BPF program into hardware.
+	XDPOffloadMode
+)
+
+type XDPOptions struct {
+	// Program must be an XDP BPF program.
+	Program *ebpf.Program
+	// Interface is string name of interface or index (int)
+	// to attach program to.
+	Interface interface{}
```

Comment:
I think `interface{}` isn't warranted in this case. If you want to be flexible I would add `InterfaceName` and `InterfaceIndex` and document that `InterfaceName` takes precedence over index if not empty.